src/tool_writeout.c

```diff
@@ -262,6 +364,33 @@ static int writeString(FILE *stream, const struct writeoutvar *wovar,
         valid = true;
       }
       break;
+    case VAR_INPUT_URLHOST:
+    case VAR_INPUT_URLPATH:
+    case VAR_INPUT_URLSCHEME:
```

Comment:
Minor drive-by nitpick: would you mind sorting the branches of the switches in `writeString` here and in `urlpart` above, so that the order is the same as in array `static const struct writeoutvar variables[]` and in the enum in `tool_writeout.h`?

Reply:
Sensible, yes.